File: src/python/main/other/rich_json_configuration.py
from ..helper.rich_json_helper import merge_into_target

_RICH_JSON_CONFIG = {
    "infoEnabled": True,
    "debugEnabled": False,
    "stringInterpolationsEnabled": True,
    "fileCacheEnabled": True,
    "lateConstructorEnabled": True,
    "crashOnNestedCloneEnabled": True,
}


def update_configuration(config):
    """
    Updates the global configuration for the library.

    :param config: The configuration dictionary.
                   - debugEnabled (bool): When true, enables rich JSON logging for debugging purposes.
                   - fileCacheEnabled (bool): Toggles the internal file system caching mechanism.
                   - lateConstructorEnabled (bool): Enables delayed object construction to improve initial load performance.
                   - crashOnNestedCloneEnabled (bool): If true, the library will throw an error when attempting to clone inside a clone structure.
    """
    global _RICH_JSON_CONFIG

    if not config or not isinstance(config, dict):
        return

    default_config = {
        "infoEnabled": True,
        "debugEnabled": False,
        "stringInterpolationsEnabled": True,
        "fileCacheEnabled": True,
        "lateConstructorEnabled": True,
        "crashOnNestedCloneEnabled": True,
    }

    _RICH_JSON_CONFIG = merge_into_target(config, default_config)

File: src/js/main/other/RichJsonConfiguration.js
import {mergeIntoTarget} from "../helper/RichJsonHelper.js";

export let __RICH_JSON_CONFIG = {
    infoEnabled: true,
    debugEnabled: false,
    stringInterpolationsEnabled: true,
    fileCacheEnabled: true,
    lateConstructorEnabled: true,
    crashOnNestedCloneEnabled: true,
}

/**
 * Updates the global configuration for the library.
 * * @param {Object} config - The configuration object.
 * @param {boolean} [config.infoEnabled=true] - When true, enables rich JSON logging.
 * @param {boolean} [config.debugEnabled=false] - When true, enables rich JSON logging for debugging purposes.
 * @param {boolean} [config.stringInterpolationsEnabled=true] - Toggles whether string interpolations are resolved.
 * @param {boolean} [config.fileCacheEnabled=true] - Toggles the internal file system caching mechanism.
 * @param {boolean} [config.lateConstructorEnabled=true] - Enables delayed object construction to improve initial load performance.
 * @param {boolean} [config.crashOnNestedCloneEnabled=false] - If true, the library will throw an error when attempting to clone inside a clone structure.
 * * @example
 * updateConfiguration({
 *  debugEnabled: true,
 *  fileCacheEnabled: false
 * });
 */
export function updateConfiguration(config) {
    if (!config || typeof config !== 'object') {
        return;
    }

    __RICH_JSON_CONFIG = mergeIntoTarget(config, {
        infoEnabled: true,
        debugEnabled: false,
        stringInterpolationsEnabled: true,
        fileCacheEnabled: true,
        lateConstructorEnabled: true,
        crashOnNestedCloneEnabled: true,
    });
}
